runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.cc

```diff
@@ -183,15 +183,14 @@ int AMDAIEDeviceModel::rows() const {
   if (device == AMDAIEDevice::xcvc1902 || device == AMDAIEDevice::xcve2802)
     return MLIRAIELegacy::rows(*this);
   assert(isNPUDevice(device) && "expected NPU device");
-  return configPtr.NumRows;
+  return devInst.NumRows;
 }
 
 int AMDAIEDeviceModel::columns() const {
   if (device == AMDAIEDevice::xcvc1902 || device == AMDAIEDevice::xcve2802)
     return MLIRAIELegacy::columns(*this);
   assert(isNPUDevice(device) && "expected NPU device");
-
-  return configPtr.NumCols;
+  return devInst.NumCols;
```

Comment:
Just FYI: `configPtr.NumCols` is the physical number of columns in the array while `devInst.NumCols` is the number in the partition.